### PR TITLE
Add the ability to show a Cesium3DTileset on only the left or right side of the splitter, for visual comparison

### PR DESCRIPTION
**Pull request by kring**
Imagery Layers have a nice feature, added in #4858, where you can display them only one side or the other of a screen-space splitter, making it really easy to do visual comparisons of two imagery layers.  This was hugely popular in TerriaJS, and probably in lots of other applications as well. And not long after that PR was merged, people started asking for the ability to do the same with other types of things.

Just five short years later, this PR adds support for splitting 3D Tiles. And it adds a bit of infrastructure that should make it pretty easy to add the feature to other primitives, too.

Here's a link to the new Sandcastle demo that uses the feature to compare Melbourne Photogrammetry with Cesium OSM Buildings:
http://cesium-dev.s3-website-us-east-1.amazonaws.com/cesium/split-primitives/Apps/Sandcastle/index.html?src=3D%20Tiles%20Compare.html

And here's another (not included as a default Sandcastle app) that compares "New York City 3D Buildings" with Cesium OSM Buildings (zoom in toward Manhattan):
[Long Sandcastle Link](http://cesium-dev.s3-website-us-east-1.amazonaws.com/cesium/split-primitives/Apps/Sandcastle/index.html#c=tVVtT9swEP4rtzKkdAOniKFppaDx0g0koIgW+FIJgnOhFo4d2U5LQfz32XFSUt6FtFZq6vNzz909d3aoFNrAmOEEFWyAwAnsoGZ5Ss4KWzBs0GK9I4WJmEA1bDTXh2IoaOHJMTHWzxMQTVEgyRRLmWFj1CSK42AooM7rH6u7A8ZRownu3T5Arni7guxLcYJa5ooiSZRMt7QF7sfBz7XVH6vNJefw0BwKn4fLgOiMM7PLFFLDpLAJlUz9OTs56P4Z1JJX7Hr0XvYlEVUYGezpdDtnPGbiWgdNF7+g+Gj4k/2/ez5+GfFOynQgA1eCLyYMoT8VFMwIIZOaFXQyKdaas9g2acLMyK8d+QxV1VSiNiCWNE9RGHKNpsvR/d2eWhGHDQ/xfZwrvWA8rsJuOJ09lsgksS04cM0OK1sWKctZUpeIcxabUU3hUSRi/nSy+lZMFP0sotgdW989Dwo8b9VVA6kc4xZ1rbD+ScQ1FltJLrzMbj9wPy6BJhSDxBIIvjw6llYAhSZXYr0YHUcCUI4A8sgBe0X6Nk7FR1DElRTkdv3R4wWRAILnOn1/wt2E0GM/IN8MpM2UIykP2eXX+5VWi7Tg23wSD4uXhc9bvZz3sPBChbI9xB0vkeVmqxA2mCkclPrNdcKo3DXiYem1fg6mGRZH7WK3d37k+vkf4xzvH+3sXfQHWycDPzmvxHLUbzMd9k773YvD3ln3jZTfp/EJVTSfrL2a9o+IfHr8aYk/GMaX1D3adYEaS41OMZabfpwBfrM0k8q4KzwgJDSYZnbwUYdXOb1BQ6jWXgoPXyivqPvKALNLrA3RlZY8N7j+uOmmvw1rrcWazcisDa3stma6iujNtZK5iJep5FK1YSFedd8aZuLOl+Wacxyhu8PbYM9WPcTdMhMx3rbhl/3M7A/P6miPrKJz1dBcaRcfJ8vKqnqHdW/36IR1ATsxGwOLN1540wLlkdZ2J8k571umYWOm+qNbdZ9vdkJrq1jr/2dQLiP38urZjHk0dS6jlc0DbySEdEK7fNnTSMmvoqdR/gE)

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @kring!
* :heavy_check_mark: Signed CLA found.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.

**Comment by ggetz** [comment]
Awesome @kring ! Does this include any tile culling optimizations like what exists for clipping planes?

**Comment by kring** [comment]
> Does this include any tile culling optimizations like what exists for clipping planes?

No, it doesn't currently. It's probably not too hard to do, as long as off-center frustum culling works correctly, which I think it does. It _might_ make for a worse user experience, though. The way this is typically used is the user drags the split point back and forth across the screen. If tiles at the split point initially pop in at lower detail, or if there's any frame rate drop (e.g. on slower systems) while dragging the split point, that won't be so good. That might be just an excuse, but I'd probably suggest treating it as a separate feature/PR in any case.

File: Source/Scene/Model.js [hunk 1]
```diff
@@ -537,6 +540,18 @@ function Model(options) {
    */
   this.showOutline = defaultValue(options.showOutline, true);
 
+  /**
+   * The {@link SplitDirection} to apply to this model.
+   *
+   * @type {SplitDirection}
+   * @default {@link SplitDirection.NONE}
+   */
+  this.splitDirection = defaultValue(
```
Comment (ggetz) [inline Source/Scene/Model.js:549]: Should this property also be added to `ModelExperimental` at this point? CC @lilleyse?

Comment (lilleyse) [reply Source/Scene/Model.js:549]: Sure, that would be helpful. Otherwise we can add it to https://github.com/CesiumGS/cesium/issues/9520

**Pull request merged**